### ISSUE TITLE
Fix failing Elasticsearch 7 tests

### ISSUE THREAD

**Issue by alexander-schranz** [handcraftedinthealps/ElasticsearchBundle#20]
Currently the Test Setup is created for Elasticsearch 5 which supports multiple Types and other features which where removed in newer Versions.

We did introduce a tests with ES 7 in: https://github.com/handcraftedinthealps/ElasticsearchBundle/pull/19

They currently will fail because of this kind of features. We would need to investigate here some time to decide which tests are needed for ES7 and which are ES5 only tests.

**Comment by toooni** [comment]
How long do you think you will need to support Elasticsearch 5 in new releases? elasticsearch/elasticsearch 5.x seems to only support PHP ^7.0, which is EOL for almost two years now.

EDIT: Sorry, my fault. ^7.0 means also PHP 7.4 ;) But Elasticsearch 5.6.x is EOL since 2019-03-11.

**Comment by alexander-schranz** [comment]
@toooni At current state aslong as it would be possible to support a wide range of elasticsearch versions we will try to support it. It would be more maintenance work to have 2 different versions of the bundle to support and keep up2date with PHP versions and Symfony versions, as I know that there are still Projects using ES5. At current state it seems like there are more projects using ES5 then ES6 and ES7

See the graphics at the bottom with the major version filter:

 - https://packagist.org/packages/handcraftedinthealps/elasticsearch-dsl/stats#major/5
 - https://packagist.org/packages/handcraftedinthealps/elasticsearch-dsl/stats#major/6
 - https://packagist.org/packages/handcraftedinthealps/elasticsearch-dsl/stats#major/7

**Comment by toooni** [comment]
I agree with the idea, but since elasticsearch/elasticsearch for ES5 doesn't support PHP8, there is not much in 5.x to maintain anymore. No new PHP Versions which will support using Elasticsearch 5.x. This ended with PHP7.4.
New features do not make much sense to implement for this version. The only case where changes for 5.x would be needed is bugfixes. But this bundle at the current state is VERY stable IMO.

**Comment by alexander-schranz** [comment]
Yeah there is currently no active development for features, I think we will decide the supported versions of PHP, Symfony and ES when a new Symfony or ElasticSearch Major Version is released and how much work it would then to support old versions.

**Comment by toooni** [comment]
Symfony 6 will arrive in November. PHP 7.4 EOL will be in November too. So, chances are that there won't be PHP7.4 support in Symfony 6. We'll see. I agree that there is no need to rush things right now. I'd recommend to wait with resolving this issue then.
